tests/Microsoft.Identity.Test.Unit/PublicApiTests/TelemetryClientTests.cs

```diff
@@ -0,0 +1,146 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Identity.Client;
+using Microsoft.Identity.Client.TelemetryCore;
+using Microsoft.Identity.Test.Common.Core.Helpers;
+using Microsoft.Identity.Test.Common.Core.Mocks;
+using Microsoft.Identity.Test.Common.Mocks;
+using Microsoft.IdentityModel.Abstractions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Microsoft.Identity.Test.Unit.PublicApiTests
+{
+    [TestClass]
+    public class TelemetryClientTests : TestBase
```

Comment:
Additional test cases for the public API could be:
`.WithTelemetryClient()` => Empty params array - should work
`.WithTelemetryClient(null)` => should throw ArgumentNullEx
`.WithTelemetryClient(telemetryClient, null)` => should throw ArgumentNullEx